lib/semian/circuit_breaker.rb

```diff
@@ -122,5 +123,19 @@ def log_state_transition(new_state)
     def disabled?
       ENV['SEMIAN_CIRCUIT_BREAKER_DISABLED'] || ENV['SEMIAN_DISABLED']
     end
+
+    def maybe_with_half_open_resource_timeout(resource)
```

Comment:
This is the coupling I think is ugly; passing in the resource and figuring out whether it quacks correctly here. When I do a 2nd iteration here with comments, I'll likely refactor to pass the raw resource into the initializer of the bulkhead and circuit—which seems a bit more clean.